docs/wiki/99 - Developers/Creating a Module.md

```diff
@@ -0,0 +1,169 @@
+│
+├── jsx/        // Original React files (JSX)
+│
+├── php/        // PHP classes
+│
+├── test/       // Test Plan and Integration Tests for the current module
+│
+├── README.md   // Brief description of what the module is and what it does
+| 
+├── .gitignore  // Local files to exclude from git. Should contain compiled JS files: `js/*`
+│
+```
+
+The filepaths below will reference these directories.
+
+_Older modules may contain other folders such as `templates/` and `ajax/`. These folders are deprecated and should
+not be used for new modules._
+
+### PHP Files
+
+The following files are required for the module to load in LORIS:
+
+* A PHP file called "module": `php/module.class.inc`.
+* A PHP file with the same name as the module: `php/my_new_module.class.inc`.
+
+The remaining PHP files will depend heavily on the kind of module you are creating. The most basic types
+of LORIS modules are "Menu Filters" and "Forms".
+
+#### Menu Filters
+
+This is a type of module that displays a table of data with options to filter the values of this table.
+
+The correct way to create a menu filter is to include additional classes that represent the idea of a
+row in the table displayed to a user. This differs from a row in the database itself; for example, a row in the 
+menu filter module may represent a joining of columns across many database tables.
+
+If you are creating a Menu Filter module, your class `php/my_new_module.class.inc` should extend
+the class `\DataFrameworkMenu`.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicom_archive.class.inc).
+
+##### Data Instance
+You should create a class `php/mynewmodulerow.class.inc` to represent a Menu Filter row. This will
+provide basic structure for the row and include permission flags that can restrict data to users
+affiliated with a given Site or Project.
+
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverow.class.inc).
+
+##### Row Provisioner.
+This file should be accompanied by a class `php/mynewmodulerowprovisioner.class.inc`. This file should
+contain the SQL statement used to query the database for data that will be consolidated into a
+Menu Filter row.
+
+[See this file for an example](../../../modules/dicom_archive/php/dicomarchiverowprovisioner.class.inc).
+
+#### Forms
+
+This type of module represents an HTML form and is used to submit data to the back-end.
+
+### JavaScript files
+
+Modern LORIS modules use ReactJS to present data. The content of these files will vary significantly
+depending on what functionality you want to provide. The best way to learn about creating JavaScript
+for a new module is by studying the code of an existing module that is similar to what you are
+trying to design.
+
+Be sure to familiarize yourself with our [ReactJS guidelines](./ReactJS Guidelines.md).
+
+### SQL Files
+
+New modules must be inserted into the database. You must create new permissions for the module and insert
```

Comment:
a little bit redundant with what follow in `Module permissions`

Reply:
Module Permissions is a subheading of this section. It's intended to expand upon the general idea of this sentence.

Reply:
This section (specially the 2nd phrase) contain details that should be in sub-section.

I would change this section to something like:
```
A pull request to create a new module must contain SQL files to let LORIS detect the module and create appropriate permissions.
```
Then in the `Modules table` below, introduce the `modules` table.